apps/arweave/src/ar_http_iface_middleware.erl

```diff
@@ -2008,21 +2008,26 @@ handle_get_chunk(OffsetBinary, Req, Encoding) ->
 								{Packing, ok};
 							{{true, _}, _StoreID} ->
 								{ok, Config} = application:get_env(arweave, config),
-								case lists:member(pack_served_chunks, Config#config.enable) of
-									false ->
-										{none, {reply, {404, #{}, <<>>, Req}}};
-									true ->
-										ok = ar_semaphore:acquire(get_and_pack_chunk,
-												infinity),
-										{RequestedPacking, ok}
+								IsPackServedChunks = lists:member(pack_served_chunks, Config#config.enable),
+								Peer = cowboy_req:peer(Req),
+								IsLocalPeerAddr = lists:foldl(fun
+									(_, true) -> true;
+									(LocalPeer, _) -> is_matching_addr(Peer, LocalPeer)
+								end, false, Config#config.local_peers),
+								case {IsPackServedChunks, IsLocalPeerAddr} of
+									{true, true} ->
+										ok = ar_semaphore:acquire(get_and_pack_chunk, infinity),
+										{RequestedPacking, ok};
+									{_, _} ->
+										{none, {reply, {403, #{}, <<>>, Req}}}
```

Comment:
I think this should be a 404 to match the old behavior: https://github.com/ArweaveTeam/arweave/pull/637/files#diff-0b7084531612365a22d52032a422eb664adc91b34cefb5f8d9499584072aa489L2013

We could have something like {true, false} be a 403

But I think best to just always 404 if the server can't or is unwilling to serve the chunk in the requested format

Reply:
Ah yes, I did this change to distinct this particular response from others during the tests. Will revert.